src/users/usersReducer.js

```diff
@@ -30,7 +31,13 @@ export default (state: UsersState = initialState, action: Action): UsersState =>
       return state; // TODO
 
     case EVENT_USER_UPDATE:
-      return state; // TODO
+      // Action will not be dispatched for a user that is not already
+      // in state.
```

Comment:
We don't want to trigger `replaceItemInArray`'s logic for when it doesn't find the item to be replaced. (It would push a malformed item to the end of the array.) This is meant to say that we don't have to check again for this; it would be redundant. But it is rather important that we get this right, so my next revision does have the reducer do a once-over on the state just to be sure.

Reply:
I see, got it.

How about not using `replaceItemInArray`? If there's sure to be a matching user, then its behavior simplifies to just a `map`. An explicit `map` here seems like less work to understand than reading through the implementation of `replaceItemInArray` (which is basically what's needed to fully understand code that uses it, because it's not documented); and there wouldn't need to be this additional case or the comment explaining it.